services/order.py

```diff
@@ -0,0 +1,34 @@
+from django.db import transaction
+from django.db.models import QuerySet
+
+from db.models import User, Order, MovieSession, Ticket
+
+
+def create_order(tickets: list[dict],
+                 username: str,
+                 date: str = None) -> None:
+
+    with transaction.atomic():
+
+        user = User.objects.get(username=username)
+        order = Order.objects.create(user=user)
+
+        for ticket in tickets:
+            movie_session = MovieSession.objects.get(
+                id=ticket["movie_session"]
+            )
+            Ticket.objects.create(row=ticket["row"],
+                                  seat=ticket["seat"],
+                                  movie_session=movie_session,
+                                  order=order)
+        if date:
+            order.created_at = date
+            order.save()
```

Comment:
created_at generates automatically. No need in this condition
Also now you do save() only if there is a date. You should do this anyway

Reply:
```This method should create an order with the provided user, if date is provided, set created_at to this date```
So my class create datetime automatically
```class Order(models.Model):
    created_at = models.DateTimeField(auto_now_add=True)```
but I rewrite it with this statement
```        if date:
            order.created_at = date
            order.save()
```
in case if I have date as argument. If date=None instance form here ```order = Order.objects.create(user=user)``` will not required save() and will be ignored and order will be just created by method Order.objects.create(user=user) witch is not required save()